test/hotspot/jtreg/compiler/intrinsics/math/TestCopySignIntrinsic.java

```diff
@@ -0,0 +1,144 @@
+
+    public static void main(String[] args) {
+        new TestFramework(TestCopySignIntrinsic.class).run();
+    }
+
+    public final int SIZE = 1024;
+    public float [] fmagnitude;
+    public float [] fsign;
+    public float [] afresult;
+    public float [] efresult;
+
+    public double [] dmagnitude;
+    public double [] dsign;
+    public double [] adresult;
+    public double [] edresult;
+
+    public TestCopySignIntrinsic() {
+        fmagnitude = new float[SIZE];
+        fsign = new float[SIZE];
+
+        dmagnitude = new double[SIZE];
+        dsign = new double[SIZE];
+
+        afresult = new float[SIZE];
+        efresult = new float[SIZE];
+
+        adresult = new double[SIZE];
+        edresult = new double[SIZE];
+
+        IntStream.range(0, SIZE - 8).forEach(i -> { fmagnitude[i] = rd.nextFloat(-Float.MAX_VALUE, Float.MAX_VALUE); });
+        IntStream.range(0, SIZE - 8).forEach(i -> { dmagnitude[i] = rd.nextFloat(-Float.MAX_VALUE, Float.MAX_VALUE); });
+        IntStream.range(0, SIZE).forEach(i -> { fsign[i] = rd.nextFloat(-Float.MAX_VALUE, Float.MAX_VALUE); });
+        IntStream.range(0, SIZE).forEach(i -> { dsign[i] = rd.nextFloat(-Float.MAX_VALUE, Float.MAX_VALUE); });
+
+        fmagnitude[SIZE - 1] = Float.NaN;
+        fmagnitude[SIZE - 2] = Float.NaN;
+        fmagnitude[SIZE - 3] = 0.0f;
+        fmagnitude[SIZE - 4] = -0.0f;
+        fmagnitude[SIZE - 5] = Float.NEGATIVE_INFINITY;
+        fmagnitude[SIZE - 6] = Float.POSITIVE_INFINITY;
+        fmagnitude[SIZE - 7] = -Float.MIN_VALUE;
+        fmagnitude[SIZE - 8] = Float.MIN_VALUE;
+
+        dmagnitude[SIZE - 1] = Double.NaN;
+        dmagnitude[SIZE - 2] = Double.NaN;
+        dmagnitude[SIZE - 3] = 0.0;
+        dmagnitude[SIZE - 4] = -0.0;
+        dmagnitude[SIZE - 5] = Double.NEGATIVE_INFINITY;
+        dmagnitude[SIZE - 6] = Double.POSITIVE_INFINITY;
+        dmagnitude[SIZE - 7] = -Double.MIN_VALUE;
+        dmagnitude[SIZE - 8] = Double.MIN_VALUE;
+
+        for (int i = 0; i < SIZE; i++) {
+            efresult[i] = Math.copySign(fmagnitude[i], fsign[i]);
+            edresult[i] = Math.copySign(dmagnitude[i], dsign[i]);
+        }
+    }
+
+    @Test
+    @IR(counts = {IRNode.COPYSIGN_F, " >0 ", IRNode.COPYSIGN_VF, " >0 "}, applyIfCPUFeature = { "avx", "true"})
+    public void testCopySignF() {
+        for (int i = 0; i < SIZE; i++) {
+            afresult[i] = Math.copySign(fmagnitude[i], fsign[i]);
+        }
+    }
+
+    @Check(test = "testCopySignF")
+    public void checkCopySignF() {
+        for (int i = 0; i < SIZE; i++) {
+            if (afresult[i] != efresult[i]) {
+                if (Float.isNaN(afresult[i]) ^ Float.isNaN(efresult[i]))  {
+                    throw new RuntimeException("Incorrect result, Math.copySign(" + fmagnitude[i] + " , " + fsign[i] + ") => " +
+                                               "expected : " + efresult[i] + " != " + " actual : " + afresult[i]);
+                }
+            }
+        }
```

Comment:
Verify.checkEQ should do this for you.... though maybe you'd have to wait for https://github.com/openjdk/jdk/pull/24224 not to get into trouble with different NaN encodings.